### PR TITLE
feat: Integrate design tokens into Tailwind configuration

### PR DESCRIPTION
**Pull request by ChengYiLin**
## Why need this change? / Root cause:

Integrate designer-provided design tokens into the project's Tailwind config for easier future development

- close : https://github.com/thementorship-tw/thementorship.tw/issues/4
- token from designer : [Notion](https://www.notion.so/Design-Token-1613af684bf880aa9e36d70db972e487?pvs=4)

## Demo

Demo is on HomePage now

https://github.com/user-attachments/assets/d49971a2-db43-4cd3-b267-cb99b51471a5

## Changes made:

- `token.json` 
  - I used the [Token Studio](https://tokens.studio/) Figma plugin to generate this JSON structure.
  - If you want to check it out, you can create a new Figma file and install the plugin there. This way, we won't mess with the designer's current work.
  
   https://github.com/user-attachments/assets/22c29929-f593-44a3-8e25-91c5d504d681
  
- `transferTokenToTailwindStyles.ts` & `tailwind.config.ts`
  - Transform JSON data into a Tailwind CSS configuration format
  - see  https://tailwindcss.com/docs/configuration

## Test Scope / Change impact:

Tailwind Config Usage Guide

- Breakpoints
  | **Class Prefix** | **Min-Width** | **Description**        |
  |------------------|---------------|------------------------|
  | --           |  0                 |  phone       |
  | `md:`    | `600px`       | tablet        |
  | `lg:`       | `840px`      | desktop  (lg)  |
  | `xl:`        | `1280px`      | desktop (xl)   |

- Typography
  - Use `text-{class}` for headings and body styles:
    - Headings: text-h1 to text-h6
    -  Subtitles: text-subtitle-{lg|md}.
    - Body text: text-body-{lg|md|sm}.

- Font Size
  - Use `text-{size}` for various font sizes:
  - ex: text-1 to text-9.

- Font Weight
  - Use `font-{weight}` to apply different font weights:
  - ex: font-bold, font-medium, font-regular, font-light.
	
- Line Height
  - Use `leading-{number}` for line height:
  - ex: leading-1 to leading-11.

- Spacing
  - it is a number to define spacing:
    ex: `pl-{number}` for horizontal padding.

- Border Radius
  - Use `rounded-{style}` for border radius:
  - ex: rounded-1 to rounded-4.

- Box Shadow
  - Use` shadow-{type}` for different shadow effects:
  - ex: shadow-action, shadow-fixed, shadow-raise, shadow-overlay.

## TODO
- `iconSize` : We need to confirm the source of our icon and the way to use it

### CODE REVIEW

**Comment by kylemocode** [comment]
For the folder name, I prefer `design-system` instead of  using camel case `designSystem`.
My thought is that I've seen more best practices using kebab-case in the past. Additionally, components are usually named in PascalCase (e.g., ProfileCard). I feel that using kebab-case might better distinguish it from components.

What are your opinions ?

**Comment by ChengYiLin** [comment]
I think using kebab-case for naming design system folder is a good choice. 
I also tend to use PascalCase for naming components,  
so this approach does make it easier to distinguish between them.  

[10d5f5f](https://github.com/thementorship-tw/thementorship.tw/pull/21/commits/10d5f5f858d99e6c8ffec1eb17fb3827b5482afa)

**Comment by kylemocode** [comment]
I think the integration is very great, I learn a lot from it, thanks!
But I have an idea, from my perspective, the demo page is a very good material, but we might need to remove it when we start to develop the main page.

How about we can create a page under a folder call `dev-only`, and use the nextjs middleware to restrict that only in dev mode can we see that page, otherwise it will be redirect to main page. Here is the example code for the middleware

```typescript
import { NextResponse } from 'next/server'
import type { NextRequest } from 'next/server'

export const config = {
  matcher: '/dev-only/:path*'
}

export function middleware(request: NextRequest) {
    // Block access if not in development
    if (process.env.NODE_ENV !== 'development') {
      return NextResponse.redirect(new URL('/', request.url))
    }

  return NextResponse.next()
}
```

Want to know your opinion or concern @ChengYiLin @gongtzuuuu

**Comment by ChengYiLin** [comment]
Indeed, the current Demo Page will be removed soon.
Creating a `dev-only` folder to preserve it seems like a good option for now 👍

---

Actually, I’ve been considering whether we should install [Storybook](https://storybook.js.org/docs) into the project to assist us in development 🤔
- On one hand, it’s truly dev-only and wouldn’t interfere with our Next.js architecture
- On the other hand, in the future, we might start developing some foundational components like Button, Input, or Dialog to ensure consistency within the project. Even though we don’t have a complete Design System, I believe Storybook could be a great tool to help us review and develop these components.

If everyone thinks it’s a good idea, we can create an issue to discuss the installation of Storybook separately.
For now, I think setting up a dev-only folder would work well.
BTW, I found a Vercel Conf talk about “[Building a Design System in Next.js with Tailwind](https://youtu.be/T-Zv73yZ_QI?feature=shared)”. 
It might be a good reference for us~

**Comment by kylemocode** [comment]
> Indeed, the current Demo Page will be removed soon. Creating a `dev-only` folder to preserve it seems like a good option for now 👍
> 
> Actually, I’ve been considering whether we should install [Storybook](https://storybook.js.org/docs) into the project to assist us in development 🤔
> 
> * On one hand, it’s truly dev-only and wouldn’t interfere with our Next.js architecture
> * On the other hand, in the future, we might start developing some foundational components like Button, Input, or Dialog to ensure consistency within the project. Even though we don’t have a complete Design System, I believe Storybook could be a great tool to help us review and develop these components.
> 
> If everyone thinks it’s a good idea, we can create an issue to discuss the installation of Storybook separately. For now, I think setting up a dev-only folder would work well. BTW, I found a Vercel Conf talk about “[Building a Design System in Next.js with Tailwind](https://youtu.be/T-Zv73yZ_QI?feature=shared)”. It might be a good reference for us~

Thanks for the sharing, we could definitely consider integrate storybook because not only the 7th will maintain this website, we hope that the website can be passed to several following mentorship programs ! Maybe we can discuss it after the 1st stage, since if we integrate now, it might slow down our development process for 1st milestone.

**Comment by ChengYiLin** [comment]
[f37306c](https://github.com/thementorship-tw/thementorship.tw/pull/21/commits/f37306c79330c72c8dea379e8988abfe3549b439)
The middleware.ts has been added to handle redirection,
and the Tailwind demo page has been moved to `/dev-only/tailwind-demo` ~

File: nextjs-app/app/dev-only/tailwind-demo/page.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
+export default async function HomePage() {
```
Comment (kylemocode) [inline nextjs-app/app/dev-only/tailwind-demo/page.tsx:1]: Can we change this component name to `TailwindDemo` or something make more sense ?

**Review by kylemocode — APPROVED** [review]

File: nextjs-app/app/dev-only/tailwind-demo/page.tsx (hunk 1 above)
Comment (ChengYiLin) [reply nextjs-app/app/dev-only/tailwind-demo/page.tsx]: Sure, my bad
[c2c5511](https://github.com/thementorship-tw/thementorship.tw/pull/21/commits/c2c55111159d39455b162046ca054fa1ce551b0f)

**Pull request merged**